### PR TITLE
Error and Audit logger webhooks

### PR DESCRIPTION
**Pull request by Alevsk**
Similar to MinIO now it's possible to configure webhooks to log all
triggered errors and incomming requests via env variables:

```
CONSOLE_LOGGER_WEBHOOK_ENABLE_<ID>
CONSOLE_LOGGER_WEBHOOK_ENDPOINT_<ID>
CONSOLE_LOGGER_WEBHOOK_AUTH_TOKEN_<ID>
CONSOLE_LOGGER_WEBHOOK_CLIENT_CERT_<ID>
CONSOLE_LOGGER_WEBHOOK_CLIENT_KEY_<ID>
CONSOLE_LOGGER_WEBHOOK_QUEUE_SIZE_<ID>

CONSOLE_AUDIT_WEBHOOK_ENABLE_<ID>
CONSOLE_AUDIT_WEBHOOK_ENDPOINT_<ID>
CONSOLE_AUDIT_WEBHOOK_AUTH_TOKEN_<ID>
CONSOLE_AUDIT_WEBHOOK_CLIENT_CERT_<ID>
CONSOLE_AUDIT_WEBHOOK_QUEUE_SIZE_<ID>
```

Signed-off-by: Lenin Alevski <alevsk.8772@gmail.com>

### CODE REVIEW

File: cmd/console/logger.go [hunk 1]
```diff
@@ -0,0 +1,290 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
```
Comment (dvaldivia) [inline cmd/console/logger.go:2]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

File: pkg/errors/errors.go [hunk 2]
```diff
@@ -0,0 +1,237 @@
+// This file is part of MinIO Console Server
+// Copyright (c) 2021 MinIO, Inc.
```
Comment (dvaldivia) [inline pkg/errors/errors.go:2]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

File: pkg/http/headers.go [hunk 3]
```diff
@@ -0,0 +1,198 @@
+// Copyright (c) 2015-2021 MinIO, Inc.
```
Comment (dvaldivia) [inline pkg/http/headers.go:1]:
```suggestion
// Copyright (c) 2022 MinIO, Inc.
```

and so many more hahaha

**Review by dvaldivia — CHANGES_REQUESTED** [review]

**Review by dvaldivia — CHANGES_REQUESTED** [review]
There's a misprint somewhere 
```
%!(EXTRA madmin.ErrorResponse=Access Denied., *errors.errorString=invalid session)
```

**Review by dvaldivia — APPROVED** [review]
Tested. LGTM.

**Review by dvaldivia — DISMISSED** [review]

**Review by bexsoft — CHANGES_REQUESTED** [review]
All terminal messages are now in the form of 

```
{"level":"INFO","errKind":"","time":"2022-04-27T22:55:33.073901Z","message":"Serving console at http://[::]:9090"}
^C{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.637599Z","message":"Shutting down... "}
{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.642138Z","message":"Stopped serving console at http://[::]:9090"}
```

Is this correct? I think people that are not using webhooks to log information will complain about using too much space
Screen Shot 2022-04-27 at 18 00 54

**Comment by Alevsk** [comment]
> ink people tha



> All terminal messages are now in the form of
> 
> ```
> {"level":"INFO","errKind":"","time":"2022-04-27T22:55:33.073901Z","message":"Serving console at http://[::]:9090"}
> ^C{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.637599Z","message":"Shutting down... "}
> {"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.642138Z","message":"Stopped serving console at http://[::]:9090"}
> ```
> 
> Is this correct? I think people that are not using webhooks to log information will complain about using too much space Screen Shot 2022-04-27 at 18 00 54

This is correct @bexsoft we are displaying more information on errors now, such as errorkind, error trace, requestID, etc

**Comment by bexsoft** [comment]
> > ink people tha
> 
> > All terminal messages are now in the form of
> > ```
> > {"level":"INFO","errKind":"","time":"2022-04-27T22:55:33.073901Z","message":"Serving console at http://[::]:9090"}
> > ^C{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.637599Z","message":"Shutting down... "}
> > {"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.642138Z","message":"Stopped serving console at http://[::]:9090"}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Is this correct? I think people that are not using webhooks to log information will complain about using too much space Screen Shot 2022-04-27 at 18 00 54
> 
> This is correct @bexsoft we are displaying more information on errors now, such as errorkind, error trace, requestID, etc

But this is happening even if I don't set any of the env vars

**Comment by Alevsk** [comment]
> > > ink people tha
> > 
> > 
> > > All terminal messages are now in the form of
> > > ```
> > > {"level":"INFO","errKind":"","time":"2022-04-27T22:55:33.073901Z","message":"Serving console at http://[::]:9090"}
> > > ^C{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.637599Z","message":"Shutting down... "}
> > > {"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.642138Z","message":"Stopped serving console at http://[::]:9090"}
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Is this correct? I think people that are not using webhooks to log information will complain about using too much space Screen Shot 2022-04-27 at 18 00 54
> > 
> > 
> > This is correct @bexsoft we are displaying more information on errors now, such as errorkind, error trace, requestID, etc
> 
> But this is happening even if I don't set any of the env vars

Yes, this is the default format for console logs now, if we print logs in json format we can even start printing logs for the embedded console  too, those logs can be parsed by any tool now

**Comment by Alevsk** [comment]
> > > ink people tha
> > 
> > 
> > > All terminal messages are now in the form of
> > > ```
> > > {"level":"INFO","errKind":"","time":"2022-04-27T22:55:33.073901Z","message":"Serving console at http://[::]:9090"}
> > > ^C{"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.637599Z","message":"Shutting down... "}
> > > {"level":"INFO","errKind":"","time":"2022-04-27T22:56:14.642138Z","message":"Stopped serving console at http://[::]:9090"}
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > Is this correct? I think people that are not using webhooks to log information will complain about using too much space Screen Shot 2022-04-27 at 18 00 54
> > 
> > 
> > This is correct @bexsoft we are displaying more information on errors now, such as errorkind, error trace, requestID, etc
> 
> But this is happening even if I don't set any of the env vars

I set now that the default is the stdout logs without any format and users can enable json/anonymous/quiet mode via environment variables :D

**Review by dvaldivia — APPROVED** [review]

**Comment by bexsoft** [comment]
Please review lint issues

**Review by bexsoft — DISMISSED** [review]

**Review by bexsoft — APPROVED** [review]

**Review by dvaldivia — APPROVED** [review]

**Pull request merged**